hilo.rb

```diff
@@ -1,34 +1,111 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@already_guessed = Array.new
+	@random_num
+	@has_won = false
+    end
+
+    # The game is started
+    def play
+	welcome
+	@random_num = get_random_num
+
+	# Gives the user max number of guesses
+	for i in 0...@max_guesses
+	    if @has_won
+		print_has_won
+		return
+	    end
+
+	    print_guesses_left(i)
+	    guess_a_number
+	end
+	
+	if @has_won
+	    print_has_won
+	    return
+	end
+	
+	print_lost_game
+    end
+
+    # Welcome message for the game
+    def welcome
+	puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
+	puts "You will have a total of #{@max_guesses} guesses.\nYou cannot guess the same number more than once."
+	puts ""	
+    end
+
+    def get_random_num
+	random_num = rand(1..100)
+    end
 
-    # only allow integer input
-    guess = gets
-    while guess.strip != guess.to_i.to_s  
-        puts "Guess a whole number!"
-        guess = gets
+    def print_guesses_left(attempt_number)
+	guesses_left = @max_guesses - attempt_number
+	puts "You have #{guesses_left} guesses left.\nPlease guess a whole number from 1 to 100 (inclusive)."
     end
-    guess = guess.to_i
 
-    # comparing randomly generated number with user guess
-    if random_num == guess
+    # Allows user to guess number and validates input by calling validate_guess
+    def guess_a_number	
+	guess = gets
+	guess = validate_guess(guess)
+	@already_guessed.append(guess)
+
+	check_guess(guess)
```

Comment:
I would move the `check_guess` up in the `play` method because as it kinda the next part of the game. Hiding the response to the user this deep in methods makes it much harder to determine where messages are coming from.

Reply:
Done all refactoring in https://github.com/amykingOW/Hi-Lo-Game/pull/4/commits/b957bd444e843ef36e7d459c28d19518097e8f45.